docs/querying/datasource.md

```diff
@@ -211,6 +219,50 @@ Inline datasources are not available in Druid SQL.
 Refer to the [Query execution](query-execution.md#inline) page for more details on how queries are executed when you
 use inline datasources.
 
+### Dynamic table append `TABLE(APPEND(...))`
```

Comment:
* Since this has the same native query syntax as `union` should this content go within the existing `union` section?
* Suggest removing the SQL syntax from the header title, so just have `#### Dynamic table append`

Reply:
I like both of those ideas. Since the native union behavior is the same, the new content can go in there, and removing the repeated `native` section won't have people wondering what the difference is.